File: backend/app/schemas/share.py
"""Project sharing Pydantic schemas — v0.4."""

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, EmailStr


class ProjectShareCreate(BaseModel):
    """Request to share a project with another user."""

    email: EmailStr
    permission: Literal["viewer", "collaborator"] = "viewer"


class ProjectShareUpdate(BaseModel):
    """Request to update share permission."""

    permission: Literal["viewer", "collaborator"]


class ProjectShareResponse(BaseModel):
    """Share record in API responses."""

    id: str
    project_id: str
    user_id: str
    username: str
    email: str
    permission: str
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/share.ts
/** Project sharing types — v0.4 */

export interface ShareCreate {
    email: string;
    permission: 'viewer' | 'collaborator';
}

export interface ShareUpdate {
    permission: 'viewer' | 'collaborator';
}

export interface ShareResponse {
    id: string;
    project_id: string;
    user_id: string;
    username: string;
    email: string;
    permission: string;
    created_at: string;
}
